linux/datadog_wrapper

```diff
@@ -1,96 +1,162 @@
-#!/usr/bin/env sh
+#!/usr/bin/env bash
 
-CURRENT_DIR=$(pwd)
+main() {
+    # Required to add the AAS metadata
+    export DD_AZURE_APP_SERVICES=1
 
-# Required to add the AAS metadata 
-export DD_AZURE_APP_SERVICES=1
+    echo "Setting application directory as $APP_DIR"
+    APP_DIR=$(pwd)
 
-if [ -z "$DD_DIR" ]; then
-    DD_DIR="/home/datadog"
-fi
+    echo "Setting Datadog environment variables"
+    setEnvVars
 
-if [ -z "$DD_BINARIES_URL" ]; then
-    DD_BINARIES_URL="https://github.com/DataDog/datadog-aas-extension/releases/download"
-fi
+    echo "Creating and switching to the Datadog directory at $DD_DIR"
+    mkdir -p "$DD_DIR" && cd "$DD_DIR" || return
 
-if [ -z "$DD_AAS_LINUX_VERSION" ]; then
-    DD_AAS_LINUX_VERSION="linux-v0.1.3-beta"
-fi
-
-setUpCommonEnv() {
-    echo "Creating Datadog directory"
-    mkdir -p $DD_DIR && cd $DD_DIR || return
-
-    FILE=trace-agent-${DD_AAS_LINUX_VERSION}.tar.gz
-
-    if [ ! -f $FILE ]; then 
-
-        if [ -f "trace-agent" ]; then 
-            echo "Removing old versions of the trace agent"
-            rm trace-agent* 
-        fi
-
-        echo "Downloading required Datadog binaries from $DD_BINARIES_URL/$DD_AAS_LINUX_VERSION/$FILE"
-        if curl -L --fail $DD_BINARIES_URL/$DD_AAS_LINUX_VERSION/$FILE -o $FILE; then echo
-            echo "Decompressing files"
-            tar -zxf $FILE || return
-        else 
-            echo "Failed to download the Datadog binary succesfully."
-            return
-        fi
-        
-    else
-        echo "Version $DD_AAS_LINUX_VERSION of the trace agent previously installed"
-    fi
+    echo "Getting the required Datadog binaries"
+    STARTUP_PIDS=()
+    getBinaries
     
-    echo "Starting trace agent"
-    ./trace-agent &
-    TRACER_PID=$!
-    trap 'kill $TRACER_PID ; exit' INT
+    case "$WEBSITE_STACK" in
+        "NODE")
+            echo "Initializing setup for a Node environment"
+            setUpNodeEnv;;
+        "DOTNETCORE")
+            echo "Initializing setup for a .NET environment"
+            setUpDotnetEnv;;
```

Comment:
Do we know what happens if the WEBSITE_STACK isn't one of these two options? Might be helpful to include a `default` option. But up to you.